CRM/Contact/Form/ContactFormTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+use Civi\API\EntityLookupTrait;
+
+/**
+ * Trait implements getContactValue + overridable getContactID functions.
+ *
+ * These are commonly used on forms - although getContactID() would often
+ * be overridden. By using these functions it is not necessary to know
+ * if the Contact ID has already been defined as `getContactID()` will retrieve
+ * them form the values available (unless it is yet to be created).
+ */
+trait CRM_Contact_Form_ContactFormTrait {
+
+  use EntityLookupTrait;
+
+  /**
+   * Get a value for the contact being acted on in the form.
+   *
+   * This can be called from any point in the form flow and if
+   * the contact can not yet be determined it will return NULL.
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function getContactValue($fieldName) {
+    if ($this->isDefined('Contact')) {
+      return $this->lookup('Contact', $fieldName);
+    }
+    $id = $this->getContactID();
+    if ($id) {
+      // GetContactID may or may not have defined this.
+      if (!$this->isDefined('Contact')) {
+        $this->define('Contact', 'Contact', ['id' => $id]);
+      }
```

Comment:
This looks wrong @eileenmcnaughton - GetContactID should definitely have defined it.

Reply:
Although @eileenmcnaughton I feel like GetContactID probably shouldn't. Feels like overloading a getter function to also be a setter isn't right. I think this function should define the contact if not defined, and GetContactID should only do what it says & get the id.

Reply:
@colemanw yeah - but that is what the function I moved is doing - I will revist once this is merged

Reply:
actually maybe not - hmm - just pondering. Yeah I will remove